### DISCUSSION TITLE
Very old package with .Net 6, with too much pain if decide to migrate.

### DISCUSSION THREAD

**Discussion by eng-mn** [serenity-is/Serenity#7125] [General]
If you tried to update it manual from 6.x.x to 8.x.x you'll face too much dirty work which will consume most of your time (this what happens with me, and I was very disappointed because I couldn't make it to get version 8.4,x works).

If you cannot update it because of VS version as you described in GitHub Issue, at least you can switch to GitHub releases (like [nopCommerce](https://github.com/nopSolutions/nopCommerce)), or user can generate new project with any chosen version of serenity from you website (like [abpframework](https://github.com/abpframework)).

this approach of stick with an old version isn't a wise choice

**Comment by VictorTomaili** [comment]
Firstly, it is not our responsibility if you choose not to upgrade until now. It's common for upgrades to take longer if the application hasn't been updated in a while. 

In version 8.x, Serenity made a transition from namespaces to ESM for TypeScript. This can bring a lot of changes. But serenity still can operate with namespaces until the migration is complete. We recommend reading the documentation for guidance.

Serenity operates on a rolling release model, which means that new applications should always use the latest version. Serene is available on the Visual Studio Marketplace, which does not offer a version history for downloads. Older versions may contain security vulnerabilities or bugs, providing another reason to keep software updated. However, some users are indifferent to these security risks.

On our website, StartSharp offers the ability to download previous versions. Not for serene. 

Should you need to use an older version, you may clone the Serene repository and check out the prior commits to build a template for that version.

**Reply by eng-mn** [reply]
Dear Victor, I think you didn't get my point.

I'm working with serenity years ago, and I have build a lot of projects (NGOs, not commercial), **Yet** I can confirm that migration isn't a good choice to rely on instead of upgrading the Serene. I don't want an old version. I just try to make it clear from one of the community member that you should not rely on manual upgrading instead of make a new release of Serene v8.x.x.

Regards,

**Reply by VictorTomaili** [reply]
https://github.com/orgs/serenity-is/discussions/7101#discussioncomment-8876323

We will release the 8.x version when possible.

**Reply by VictorTomaili** [reply]
https://github.com/serenity-is/Serenity/issues/7020#issuecomment-2081604093

**Reply by eng-mn** [reply]
Thanks so much Dear Victor, that's what we're waiting for 🙏

**Comment by AlexNolasco** [comment]
> If you tried to update it manual from 6.x.x to 8.x.x you'll face too much dirty work which will consume most of your time (this what happens with me, and I was very disappointed because I couldn't make it to get version 8.4,x works).
> 
> If you cannot update it because of VS version as you described in GitHub Issue, at least you can switch to GitHub releases (like [nopCommerce](https://github.com/nopSolutions/nopCommerce)), or user can generate new project with any chosen version of serenity from you website (like [abpframework](https://github.com/abpframework)).
> 
> this approach of stick with an old version isn't a wise choice

oh thank you for suggesting those alternatives..
